src/site/page/mod.rs

```diff
@@ -60,6 +60,22 @@ impl Page {
         })
     }
 
+    pub fn new_from_file_with_dir(source: &str, layout: &Layout, config: &Config) -> Result<Self> {
```

Comment:
could just accept syntax theme instead of all config i think?

Reply:
i figured the signature should be similar to the function above (which it is based on)

Reply:
ah word, we can leave this for a future pass on that then